packages/api/src/record/service/copy.ts

```diff
@@ -0,0 +1,81 @@
+import { getFolderByFolderLinkId } from "../../folder/service";
+import { type FolderRow, FolderStatus } from "../../folder/models";
+import { validateCreateEventRequest } from "../../event/validators";
+import type { CreateEventRequest } from "../../event/models";
+import { getRecordById } from "../service";
+import { type ArchiveRecord, type CopyRecordRequest, RecordStatus, type RecordVO } from "../models";
+import { InternalServerError } from "http-errors";
+
+export const getRecordFromRecordVo = async (
+  recordVO: RecordVO
+): Promise<ArchiveRecord | null> => {
+  // const archiveId = recordData.RecordVO.archiveNbr;
+  let record = null;
+  let recordId = null;
+
+  recordId = { recordId: recordVO.recordId.toString() };
+  [record] = await getRecordById({
+    recordIds: [recordId.recordId],
+    accountEmail: "test11@testt.com", // Where to get this email from?
+  });
+
+  if (record === undefined) {
+    throw new InternalServerError("RecordVO does not exist");
+  }
+  return record;
+};
+
+export const copyRecord = async (
+  recordData: CopyRecordRequest
+): Promise<FolderRow> => {
+  // TODO check space over - How to check if there is enough space?
```

Comment:
The `account_space` table keeps the running total of the space available to an account. The account to check is the calling account, unless the archive where the destination folder lives has `payerAccountId` set, in which case that's the account to check.